src/NFTBackedToken.sol

```diff
@@ -87,6 +91,10 @@ contract NFTBackedToken is
 
         for (uint256 i; i < tokensIn.length; ++i) {
             nft.transferFrom(msg.sender, address(this), tokensIn[i]);
+            // the spec says `tokensIn` must be owned by msg.sender. it's possible to do a swap like this:
+            // tokensIn: [0, 1], tokensOut: [1, 2]
+            // then msg.sender doesn't need to own id 1.
+            // couldn't think of a way it can do damage, but maybe I missed something?
```

Comment:
@davidbrai nice catch :) I don't think there's anything wrong with the scenario you describe here; I wrote the code this way to avoid having 2 loops, but can make it tighter.

your scenario is basically a swap from 0 to 2 thought 1; doesn't leave room for abuse IMO, but still happy to tighten it up!

Reply:
fixed now